cli/src/nonce.rs

```diff
@@ -0,0 +1,466 @@
+        return Err(CliError::BadParameter(
+            "Unable to create new nonce, no nonce account found".to_string(),
+        )
+        .into());
+    }
+
+    let ix = nonce(&nonce_account_pubkey);
+    let (recent_blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+    let mut tx = Transaction::new_signed_with_payer(
+        vec![ix],
+        Some(&config.keypair.pubkey()),
+        &[&config.keypair, nonce_account],
+        recent_blockhash,
+    );
+    check_account_for_fee(rpc_client, config, &fee_calculator, &tx.message)?;
+    let result =
+        rpc_client.send_and_confirm_transaction(&mut tx, &[&config.keypair, nonce_account]);
+    log_instruction_custom_error::<SystemError>(result)
+}
+
+pub fn process_get_nonce(rpc_client: &RpcClient, nonce_account_pubkey: &Pubkey) -> ProcessResult {
+    let nonce_account = rpc_client.get_account(nonce_account_pubkey)?;
+    if nonce_account.owner != nonce_program::id() {
+        return Err(CliError::RpcRequestError(
+            format!("{:?} is not a nonce account", nonce_account_pubkey).to_string(),
+        )
+        .into());
+    }
+    match nonce_account.state() {
+        Ok(NonceState::Uninitialized) => Ok("Nonce account is uninitialized".to_string()),
+        Ok(NonceState::Initialized(_, hash)) => Ok(format!("{:?}", hash)),
+        Err(err) => Err(CliError::RpcRequestError(format!(
+            "Account data could not be deserialized to nonce state: {:?}",
+            err
+        ))
+        .into()),
+    }
+}
+
+pub fn process_withdraw_from_nonce_account(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    nonce_account: &Keypair,
+    destination_account_pubkey: &Pubkey,
+    lamports: u64,
+) -> ProcessResult {
+    let (recent_blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+
+    let ix = withdraw(
+        &nonce_account.pubkey(),
+        destination_account_pubkey,
+        lamports,
+    );
+    let mut tx = Transaction::new_signed_with_payer(
+        vec![ix],
+        Some(&config.keypair.pubkey()),
+        &[&config.keypair, nonce_account],
+        recent_blockhash,
+    );
+    check_account_for_fee(rpc_client, config, &fee_calculator, &tx.message)?;
+    let result =
+        rpc_client.send_and_confirm_transaction(&mut tx, &[&config.keypair, nonce_account]);
+    log_instruction_custom_error::<NonceError>(result)
+}
+
+pub fn process_show_nonce_account(
+    rpc_client: &RpcClient,
+    nonce_account_pubkey: &Pubkey,
+    use_lamports_unit: bool,
+) -> ProcessResult {
+    let nonce_account = rpc_client.get_account(nonce_account_pubkey)?;
+    if nonce_account.owner != nonce_program::id() {
+        return Err(CliError::RpcRequestError(
+            format!("{:?} is not a nonce account", nonce_account_pubkey).to_string(),
+        )
+        .into());
+    }
+    match nonce_account.state() {
+        Ok(NonceState::Uninitialized) => Ok("Nonce account is uninitialized".to_string()),
```

Comment:
An uninitialized account will still have a balance that the user may be interested in